src/main/java/server/ProtocolRunner.java

```diff
@@ -31,8 +33,10 @@ public void run() {
             String input = in.readLine();
             while(input != null) {
                 Request request = requestParser.create(input);
-                Response response = router.route(request);
+                PathFinder pathFinder = new PathFinder(request, app);
```

Comment:
Are we initializing a new `PathFinder` for every request?

Reply:
Currently yes, I'll refactor it to be reusable